pkgs/tools/typesetting/sile/default.nix

```
@@ -42,56 +44,69 @@ let
   ] ++ lib.optionals (lib.versionOlder lua.luaversion "5.3") [
     compat53
   ]);
-in
 
-stdenv.mkDerivation (finalAttrs: {
+  in stdenv.mkDerivation (finalAttrs: {
   pname = "sile";
-  version = "0.14.17";
+  version = "0.15.4";
 
   src = fetchurl {
-    url = "https://github.com/sile-typesetter/sile/releases/download/v${finalAttrs.version}/sile-${finalAttrs.version}.tar.xz";
-    sha256 = "sha256-f4m+3s7au1FoJQrZ3YDAntKJyOiMPQ11bS0dku4GXgQ=";
+    url = "https://github.com/sile-typesetter/sile/releases/download/v${finalAttrs.version}/sile-${finalAttrs.version}.tar.zst";
+    sha256 = "0n13kv1pabqccjycswcfafhks54l9ck6k8dfkd436jggharkgn1m";
   };
 
-  configureFlags = [
-    "--with-system-luarocks"
-    "--with-manual"
-  ];
-
   nativeBuildInputs = [
-    gitMinimal
-    pkg-config
     makeWrapper
+    pkg-config
+    rustPlatform.cargoSetupHook
+    jq
+    cargo
+    rustc
+    zstd
   ];
+
+  # In Nixpkgs, we don't copy the Cargo.lock file from the repo to Nixpkgs'
+  # repo, but we inherit src, and specify a hash (it is a fixed output
+  # derivation). `nix-update` and `nixpkgs-update` should be able to catch that
+  # hash and update it as well when performing updates.
+  cargoDeps = rustPlatform.importCargoLock {
+    lockFile = ./Cargo.lock;
+  };
```

Ironically your code contradicts the comments and vendors `Cargo.lock`. But we don't need to vendor it here.

```suggestion
  cargoDeps = rustPlatform.fetchCargoTarball {
    inherit (finalAttrs) src;
    dontConfigure = true;
    hash = "sha256-FD2otvk92/99AT3BEmfbID8j8MFIicshcocPZalPTHQ=";
  };
```

---

That comment came from code copied out of the upstream Flake, but I had to tinker with it so much I lost track of what it was saying. I was so confused by the apparent need to vendor this (I'm glad to find it is not actually needed).

Where did you get this hash?

---

I used the trust-on-first-use method, i.e. set `hash = "";`, build, and get the hash from the error. There might be a more elegant way that I don't know of.

---

Fair enough. Because I didn't understand it I tried using the hash of the Cargo.lock file from upstream, but that turned out *not* to work and gave me this hash in the error ;-) Given that the lock file is in the upstream source tarball and there is already a checksum on that, it seems like a bit of redundant work here. Whatever this is actually hashing should be derived from the already checksumed source inputs.